### PR TITLE
[build-script] Add a LLDB PR testing preset

### PR DESCRIPTION
**Pull request by JDevlieghere**
Create an LLDB PR testing preset.

### CODE REVIEW

**Comment by JDevlieghere** [comment]
@swift-ci please test

**Review by shahmishal — APPROVED** [review]

**Comment by shahmishal** [comment]
preset=lldb-pull-request
@swift-ci Please test with preset macOS Platform

File: utils/build-presets.ini [hunk 1]
```diff
@@ -296,6 +296,16 @@ lldb-build-type=Release
 lldb-test-swift-only
 lldb-assertions
 
+[preset: lldb-pull-request]
```
Comment (adrian-prantl) [inline utils/build-presets.ini:299]: is `assertions` implicit?

Comment (JDevlieghere) [reply utils/build-presets.ini:299]: Yes, it's par of relwithdeb

**Comment by JDevlieghere** [comment]
preset=lldb-pull-request
@swift-ci Please test with preset macOS Platform

**Comment by JDevlieghere** [comment]
@swift-ci please test

**Comment by JDevlieghere** [comment]
preset=lldb-pull-request
@swift-ci Please test with preset macOS Platform

**Comment by JDevlieghere** [comment]
We're seeing one legitimate failure with the new preset: `functionalities/postmortem/elf-core/TestLinuxCore.py (also failing on https://ci.swift.org/view/LLDB/job/oss-lldb-incremental-macos-cmake/). Not sure what's up with Windows.

@swift-ci please test platform Windows

**Comment by JDevlieghere** [comment]
@swift-ci please test Windows platform

**Comment by JDevlieghere** [comment]
preset=lldb-pull-request
@swift-ci Please test with preset macOS Platform

**Pull request merged**